services/src/main/java/com/chatbot/services/pubsubservices/PubSubPullSubscriber.java

```diff
@@ -0,0 +1,134 @@
+package com.chatbot.services.pubsubservices;
+
+import java.util.Map;
+
+import com.chatbot.services.ChatServiceConstants;
+import com.chatbot.services.asyncservices.HangoutsAsyncService;
+import com.chatbot.services.protobuf.TriggerEventNotificationOuterClass.TriggerEventNotification;
+import com.chatbot.services.protobuf.TriggerEventNotificationOuterClass.TriggerEventNotification.ChatClient;
+import com.chatbot.services.protobuf.TriggerEventNotificationOuterClass.TriggerEventNotification.Event;
+import com.google.api.gax.batching.FlowControlSettings;
+import com.google.api.gax.core.ExecutorProvider;
+import com.google.api.gax.core.InstantiatingExecutorProvider;
+import com.google.cloud.pubsub.v1.AckReplyConsumer;
+import com.google.cloud.pubsub.v1.MessageReceiver;
+import com.google.cloud.pubsub.v1.Subscriber;
+import com.google.protobuf.Struct;
+import com.google.pubsub.v1.ProjectSubscriptionName;
+import com.google.pubsub.v1.PubsubMessage;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+
+// Handle messages from cloud sub using a pull subscriber
+// Annotating this class with @Component will launch the pull subscriber on application start up
+public class PubSubPullSubscriber {
+
+  private final HangoutsAsyncService asyncService;
+  private static final Logger logger = LoggerFactory.getLogger(PubSubPullSubscriber.class);
+  private static String projectID;
+  private static String subscriptionID;
+  private static Long maxOutstandingElements;
+  private static Long maxOutstandingBytes;
+  private static int threadCount;
+
+  PubSubPullSubscriber(
+      @Value("${pubsubConfig.maxOutstandingElements}") final String maxOutstandingElementsToSet,
+      @Value("${pubsubConfig.maxOutstandingBytes}") final String maxOutstandingBytesToSet,
+      @Value("${pubsubConfig.threadCount}") final String threadCountToSet,
+      final HangoutsAsyncService asyncServiceToSet) throws InterruptedException {
```

Comment:
You can use same names here, while assigning use "this."
Eg. this.asyncService = asyncService

Reply:
updated

Reply:
You just changed it for 1 argument, what about the others?

Reply:
Oh! didn't notice that, will update